cds_ils/migrator/series/api.py

```diff
@@ -66,75 +72,109 @@ def create_multipart_volumes(pid, multipart_legacy_recid, migration_volumes):
         if volume_number not in volumes:
             volumes[volume_number] = {}
         volume = volumes[volume_number]
-        for key in obj:
-            if key != "volume":
-                if key in volume:
-                    raise KeyError(
-                        'Duplicate key "{}" for multipart {}'.format(
-                            key, multipart_legacy_recid
+        if "isbn" in obj:
+            if "isbns" not in volume:
+                volume["isbns"] = []
+            volume["isbns"].append({
+                "value": obj["isbn"],
+                "is_electronic": bool(obj["is_electronic"])
+            })
+            # TODO physical description
+        elif "barcode" in obj:
+            # this entry represents an item
+            if "items" not in volume:
+                volume["items"] = []
+            volume["items"].append({
+                "barcode": obj["barcode"]
+            })
+        else:
+            for key in obj:
```

Comment:
sorry I'm not sure what is happening here and why, maybe needs a comment or more self explanatory code

Reply:
This part was there already; it's basically inner joining the volumes together by their `volume` number.